evm/utils/test_builder.py

```diff
@@ -0,0 +1,309 @@
+        account_state_dict.setdefault("balance", 0)
+        account_state_dict.setdefault("nonce", 0)
+        account_state_dict.setdefault("code", b"")
+        account_state_dict.setdefault("storage", OrderedDict())
+    return state
+
+
+def state_list_to_dict(state_list):
+    d = {}
+    for state_item in state_list:
+        d = assoc_in(d, state_item[:-1], state_item[-1])
+    return d
+
+
+def merge_states(state1, state2):
+    new_state = copy.deepcopy(state2)
+    for key, value in state1.items():
+        if key not in new_state:
+            new_state[key] = copy.deepcopy(value)
+        elif isinstance(value, Mapping) and isinstance(new_state[key], Mapping):
+            new_state[key] = merge_states(new_state[key], value)
+    return new_state
+
+
+def without_duplicates(l):
+    return list(OrderedDict.fromkeys(l))
+
+
+def flatten(l):
+    return list(itertools.chain.from_iterable(l))
+
+
+def int_to_hex(i):
+    return encode_hex(int_to_big_endian(i))
+
+
+def get_version_from_git():
+    version = subprocess.check_output(["git", "describe"]).strip()
+    return force_text(version)
+
+
+env_formatter = curried.apply_formatters_to_dict({
+    "currentCoinbase": to_checksum_address,
+    "currentDifficulty": int_to_hex,
+    "currentGasLimit": int_to_hex,
+    "currentNumber": int_to_hex,
+    "currentTimestamp": int_to_hex,
+    "previousHash": encode_hex,
+})
+
+storage_item_formatter = combine_argument_formatters(int_to_hex, int_to_hex)
+
+
+def storage_formatter(storage_dict):
+    items = list(storage_dict.items())
+    formatted_items = apply_formatter_to_array(storage_item_formatter, items)
+    return dict(formatted_items)
+
+
+account_state_formatter = curried.apply_formatters_to_dict({
+    "balance": int_to_hex,
+    "nonce": int_to_hex,
+    "code": encode_hex,
+    "storage": storage_formatter,
+})
+
+
+state_item_formatter = combine_argument_formatters(to_checksum_address, account_state_formatter)
+
+
+def state_formatter(state_dict):
+    items = list(state_dict.items())
+    formatted_items = apply_formatter_to_array(state_item_formatter, items)
+    return dict(formatted_items)
+
+
+class StateTest:
+
+    def __init__(
+        self,
+        name,
+        *,
+        comment="",
+        filled_with=None,
+        forks=ALL_FORKS,
+        env=None,
+        pre_state=None,
+        transaction,
+        secret_key,
+        output=None,
+        state_changes=None,
+        logs=None
+    ):
+        # meta data
+        self.name = name
+        self.comment = comment
+        if filled_with is not None:
+            self.filled_with = filled_with
+        else:
+            self.filled_with = FILLED_WITH_TEMPLATE.format(version=get_version_from_git())
+        self.info = OrderedDict([
+            ("comment", self.comment),
+            ("filledwith", self.filled_with),
+        ])
+        self.forks = forks
+
+        # pre state
+        self.env = merge(DEFAULT_ENV, env or OrderedDict())
```

Comment:
You could just change `OrderedDict()` to `{}` here for slightly improved readability.  Same with other places that you use an empty ordered dict (unless your relying on the underlying `merge` to return an `OrderedDict` at which point please ignore (does it do that?).

Reply:
Hmm, I assumed it does, but now I checked and I'm wrong... I tried to use `OrderedDict`s to keep a sensible order in the final JSON file (pre before transaction before post). But apparently the `ethereum/tests` standard is ordering alphabetically, so I don't think it's worth the effort. I'm gonna switch to plain dictionaries.

Reply:
In this same vein, we *should* use `sort_keys=True` if this actually ends up dumping JSON strings at any point.